packages/block-library/src/navigation/edit/index.js

```diff
@@ -186,7 +186,7 @@ function Navigation( {
 				hasSubmenus: !! innerBlocks.find(
 					( block ) => block.name === 'core/navigation-submenu'
 				),
-				hasUncontrolledInnerBlocks: _hasUncontrolledInnerBlocks,
+				hasUncontrolledInnerBlocks: !! _hasUncontrolledInnerBlocks,
```

Comment:
Why the `!!`? Could we fix `_hasUncontrolledInnerBlocks` to be definitely bool?

Reply:
These values are derived from values which a coerced to Booleans but are in fact empty arrays or numbers.

I just refer the clarity of `hasX` definitely being a Boolean rather than some other value.

> Could we fix _hasUncontrolledInnerBlocks to be definitely bool?

We'd just end up with

```js
const _hasUncontrolledInnerBlocks =
				!! _uncontrolledInnerBlocks?.length;
```

Reply:
✅ Fixed in https://github.com/WordPress/gutenberg/pull/38907/commits/092947d23a0bc2d8ac260b5d808a84b24af35403